test/cli/whole-program_test.py

```diff
@@ -214,6 +216,65 @@ def test_suppress_inline_project_builddir_j(tmpdir):
     os.mkdir(build_dir)
     __test_suppress_inline_project(tmpdir, ['-j2', '--cppcheck-build-dir={}'.format(build_dir)])
 
+@pytest.mark.parametrize("jobs,builddir", ((1,False), (1,True), (2,False), (2,True)))
+def test_addon_rerun(tmpdir, jobs, builddir):
+    """Rerun analysis and ensure that misra CTU works; with and without build dir, with -j1 and -j2"""
+    args = [
+        f'-j{jobs}',
+        '--addon=misra',
+        '--enable=style',
+        '--template={id}',
+        'whole-program']
+    if builddir:
+        args.append(f'--cppcheck-build-dir={tmpdir}')
+    _, _, stderr = cppcheck(args, cwd=__script_dir)
+    assert 'misra-c2012-5.8' in stderr
+    _, _, stderr = cppcheck(args, cwd=__script_dir)
+    assert 'misra-c2012-5.8' in stderr
+
+@pytest.mark.parametrize("jobs", (1,2))
```

Comment:
No need to explicitly test `-j`. That is done automatically within the CI.

Reply:
you mean that the CI will add -j options in the args I create in the test case. I would not think so.

Reply:
That was added in #6061.

And as pointed out somewhere else this should probably be done for `--cppcheck-build-dir` as well as that should not change the behavior in single run execution. But that is for later.

Reply:
you miss the point as far as I see. I want to test with a single thread and I ALSO want to test with multiple threads. So I need to remove that TEST_INJECT environment variable if that is set.

Reply:
As far as I see I don't need to remove TEST_CPPCHECK_INJECT_J because it has this code:

    if 'TEST_CPPCHECK_INJECT_J' in os.environ:
        found_j = False
        for arg in args:
            if arg.startswith('-j'):
                found_j = True
                break
        if not found_j:
            arg_j = '-j' + str(os.environ['TEST_CPPCHECK_INJECT_J'])
            args.append(arg_j)

so TEST_CPPCHECK_INJECT_J should have no effect if explict -j option is used as far as I see.

Reply:
:+1: I have updated